pkg/secrets/secretsdb.go

```diff
@@ -0,0 +1,197 @@
+package secrets
+
+import (
+	"encoding/json"
+	"io/ioutil"
+	"os"
+	"time"
+
+	"github.com/pkg/errors"
+)
+
+type db struct {
+	// Secrets maps a secret id to secret metadata
+	Secrets map[string]Secret `json:"secrets"`
+	// NameToID maps a secret name to a secret id
+	NameToID map[string]string `json:"nametoid"`
+	// IDToName maps a secret id to a secret name
+	IDToName map[string]string `json:"idtoname"`
+	// lastModified is the time when the database was last modified on the file system
+	lastModified time.Time
+}
+
+// loadDB loads database data into memory if it has been modified
+func (s *SecretsManager) loadDB() error {
+	fileInfo, err := os.Stat(secretsFile)
```

Comment:
Please add verbose comments to this function.  It might not be immediately clear why returning `nil` is desired if the file doesn't exist.